### PR TITLE
Update mod version for genref

### PR DESCRIPTION
**Pull request by tengqm**
This also fixes the problems in generating kubeadm-config reference.
closes: #241

### CODE REVIEW

File: genref/go.mod [hunk 1]
```diff
@@ -3,31 +3,29 @@ module github.com/kubernetes-sigs/reference-docs/genref
...
 	github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect
 	github.com/pkg/errors v0.9.1
-	github.com/tengqm/kubeconfig v0.0.0-20210703133455-61479ab64f9d
-	github.com/yuin/goldmark v1.2.1
-	github.com/yuin/goldmark-highlighting v0.0.0-20200307114337-60d527fdb691
-	golang.org/x/mod v0.3.1-0.20200828183125-ce943fd02449 // indirect
-	golang.org/x/sys v0.0.0-20210225134936-a50acf3fe073 // indirect
-	golang.org/x/tools v0.1.0 // indirect
-	gopkg.in/check.v1 v1.0.0-20200227125254-8fa46927fb4f // indirect
+	github.com/tengqm/kubeconfig v0.0.0-20210707141017-7849c7b17290
```
Comment (sftim) [inline genref/go.mod:10]: How come this mod is added?

**Review by sftim — COMMENTED** [review]
- I'd prefer to see a change like this split into 2 commits (or 2 PRs): one that updates the mod version, and another to make the kubeconfig changes
- The new dependency `github.com/tengqm/kubeconfig` at version 0.0.0 looks possibly out of place; why is it included?

File: genref/go.mod (hunk 1 above)
Comment (tengqm) [reply genref/go.mod:10]: This repo was created for kubeadm config which is still embedded in the k/k repo. For generating reference for kubeadm config, we had to clone the source, eliminate its dependencies on k/k.

Comment (kbhawkey) [reply genref/go.mod:13]: 👍

**Comment by kbhawkey** [comment]
This looks like cleanup.
/kind cleanup

**Pull request merged**